pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java

```diff
@@ -99,6 +99,14 @@ public void testGeneratedQueries()
     super.testGeneratedQueries(false, true);
   }
 
+  @Test
+  public void testQueryOptions()
+      throws Exception {
+    String pinotQuery = "SELECT * FROM mytable OPTION(multiStageLeafLimit=1)";
```

Comment:
let's use `SET multistageLeafLimit = 1` as the `OPTION` usage is being deprecated.
also how come H2 and pinot return the same result? i thought specifically this option makes the leaf only return 1 row right?